sentry-kotlin-extensions/src/main/java/io/sentry/kotlin/SentryCoroutineExceptionHandler.kt

```diff
@@ -0,0 +1,26 @@
+package io.sentry.kotlin
+
+import io.sentry.IScopes
+import io.sentry.ScopesAdapter
+import io.sentry.SentryEvent
+import io.sentry.exception.ExceptionMechanismException
+import io.sentry.protocol.Mechanism
+import kotlinx.coroutines.CoroutineExceptionHandler
+import org.jetbrains.annotations.ApiStatus
+import kotlin.coroutines.AbstractCoroutineContextElement
+import kotlin.coroutines.CoroutineContext
+
+@ApiStatus.Experimental
+public open class SentryCoroutineExceptionHandler(private val scopes: IScopes = ScopesAdapter.getInstance()) :
+    AbstractCoroutineContextElement(CoroutineExceptionHandler), CoroutineExceptionHandler {
+
+    override fun handleException(context: CoroutineContext, exception: Throwable) {
+        val mechanism = Mechanism().apply {
+            type = "CoroutineExceptionHandler"
```

Comment:
we could also set `handled = true` here I guess, because this is what the handler does, right?